sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -316,10 +348,95 @@ private Message createManagementMessage(String operation, String linkName) {
         return message;
     }
 
+    /***
+     * Create a Amqp key, value map to be used to create Amqp mesage for scheduling purpose.
+     *
+     * @param messageToSchedule The message which needs to be scheduled.
+     * @return Map of key and value in Amqp format.
+     * @throws AmqpException When payload exceeded maximum message allowed size.
+     */
+    private Map<String, Object> createScheduleMessgeAmqpValue(ServiceBusMessage messageToSchedule) {
+        int maxMessageSize = MAX_MESSAGE_LENGTH_SENDER_LINK_BYTES;
+        Map<String, Object> requestBodyMap = new HashMap<>();
+        List<Message> messagesToSchedule = new ArrayList<>();
+        messagesToSchedule.add(messageSerializer.serialize(messageToSchedule));
+        Collection<HashMap<String, Object>> messageList = new LinkedList<>();
+        for (Message message : messagesToSchedule) {
+            final int payloadSize = messageSerializer.getSize(message);
+            final int allocationSize =
+                Math.min(payloadSize + MAX_MESSAGING_AMQP_HEADER_SIZE_BYTES, maxMessageSize);
+            final byte[] bytes = new byte[allocationSize];
+
+            int encodedSize;
+            try {
+                encodedSize = message.encode(bytes, 0, allocationSize);
+            } catch (BufferOverflowException exception) {
+                final String errorMessage =
+                    String.format(Locale.US,
+                        "Error sending. Size of the payload exceeded maximum message size: %s kb",
+                        maxMessageSize / 1024);
+                throw logger.logExceptionAsWarning(new AmqpException(false,
+                    AmqpErrorCondition.LINK_PAYLOAD_SIZE_EXCEEDED, errorMessage, exception, getErrorContext()));
+            }
+
+            HashMap<String, Object> messageEntry = new HashMap<>();
+
+            messageEntry.put(MESSAGE, new Binary(bytes, 0, encodedSize));
+            messageEntry.put(MESSAGE_ID, message.getMessageId());
+            messageList.add(messageEntry);
+        }
+        requestBodyMap.put(MESSAGES, messageList);
+
+        return requestBodyMap;
+    }
+
     private AmqpErrorContext getErrorContext() {
         return new SessionErrorContext(fullyQualifiedNamespace, entityPath);
     }
 
+    private Mono<Void> cancelSchedule(Long[] cancelScheduleNumbers) {
```

Comment:
This doesn't need to be a method, it is only used in one place. Roll it up into the calling method.